bin/sass.dart

```diff
@@ -52,8 +53,15 @@ Future<void> main(List<String> args) async {
       return;
     }
 
-    var graph = StylesheetGraph(
-        ImportCache(loadPaths: options.loadPaths, logger: options.logger));
+    var graph = StylesheetGraph(ImportCache(
+        loadPaths: options.loadPaths,
+        logger: DeprecationHandlingLogger(options.logger,
+            fatalDeprecations: options.fatalDeprecations,
+            futureDeprecations: options.futureDeprecations,
+            // This logger may be reused by multiple parses, so we don't want
+            // warnings from a previous parse to cause subsequent warnings to
+            // be skipped.
+            limitRepetition: false)));
```

Comment:
I have a couple questions here:

* Previously, the `Options` class was responsible for setting up the logger according to the options passed. Why not do the same with `DeprecationHandlingLogger`?

* If we don't limit repetitions here, under what circumstances _will_ we limit repetitions on the CLI?

Reply:
This wrapped logger is only used during parsing. `options.logger` is wrapped by a `DeprecationHandlingLogger` again during the actual compilation (this time, with `limitRepetition` set to true if `verbose` is false).

This matches the current behavior with the `TerseLogger` that `DeprecationHandlingLogger` replaces (repetitions aren't limited for deprecation warnings during parsing). The need to reset the state of `DeprecationHandlingLogger` is also why we don't just have `options.logger` produce it.

We could change `options.logger` to just produce a single `DeprecationHandlingLogger`, but we'd then need some way for `StylesheetGraph` and the compiler to reset its state after each run. Do you think that makes more sense instead of wrapping the base logger multiple times?

Reply:
No, I think this is fine. It might be worth documenting more of what you just said in the comments, though.